packages/backend/src/managers/podmanConnection.ts

```diff
@@ -54,6 +60,71 @@ export class PodmanConnection extends Publisher<ContainerProviderConnectionInfo[
     this.#disposables = [];
   }
 
+  /**
+   * Execute the podman cli with the arguments provided
+   *
+   * @example
+   * ```
+   * const result = await podman.execute(connection, ['machine', 'ls', '--format=json']);
+   * ```
+   * @param connection
+   * @param args
+   * @param options
+   */
+  execute(connection: ContainerProviderConnection, args: string[], options?: RunOptions): Promise<RunResult> {
+    const podman = extensions.getExtension('podman-desktop.podman');
+    if (!podman) {
+      console.warn('cannot find podman extension api');
+      return this.executeLegacy(args, options);
+    }
+
+    const podmanApi: {
+      exec(args: string[], options?: PodmanRunOptions): Promise<RunResult>;
+    } = podman.exports;
+
+    return podmanApi.exec(args, {
+      ...options,
+      connection: this.getProviderContainerConnection(connection),
+    });
+  }
+
+  /**
+   * Execute a command inside the podman machine
+   *
+   * @example
+   * ```
+   * const result = await podman.executeSSH(connection, ['ls', '/dev']);
+   * ```
+   * @param connection
+   * @param args
+   * @param options
+   */
+  executeSSH(connection: ContainerProviderConnection, args: string[], options?: RunOptions): Promise<RunResult> {
+    return this.execute(connection, ['machine', 'ssh', this.getNameLegacyCompatibility(connection), ...args], options);
+  }
+
+  /**
```

Comment:
I'm not sure we need the code for < 1.13 as the extension is already requiring API > 1.13.0

Reply:
We are not ?

https://github.com/containers/podman-desktop-extension-ai-lab/blob/35367499f94a8726bb36dbc3a36045cb4677defa/packages/backend/package.json#L11

We uses the latest api, but we don't have hard requirements on latest version, so we are _in theory_ compatible with older version up to 1.8.

Reply:
in the catalog it says minimum 1.12 for 1.2 of ai lab

https://github.com/containers/podman-desktop-catalog/blob/gh-pages/api/extensions.json#L689

and in package.json, we consume 1.13.0 snapshot API

I'm assuming some API changes (new additions) are also missing since 1.8

so consuming 1.13 API should result in 1.13 minimal requirement